### DISCUSSION TITLE
Regex is possible to add in defineRoutes path?

### DISCUSSION THREAD

**Discussion by pankaj91as** [remix-run/remix#6886] [Q&A]
I am facing an routing issue in remix v1.19.0.
I find a lot in the document, as well on network but i haven't got any solution for my routing problem.

I have two routes similer like mentioned below:
```
/world/egypts-president-grants-pardon-to-detained-researcher-patrick-zaki-617616
/world/united-stats
```

i was trying to route these two path. These two paths should open two diffrent layout from pages.
I tried something like below, please let me know is it something possible in V2?

```
  routes(defineRoutes) {
    return defineRoutes((route) => {
      route("/:category/:subcategory([a-zA-Z-]+$)", "routes/listing.tsx");
      route("/:category/:slug([a-zA-Z0-9-]+[0-9]+$)", "routes/details.tsx");
    });
  },
```
only the diffrence in these two paths is, one path having slug which is ending with numerical values.

Thanks for your help.

**Comment by kiliman** [comment]
Remix doesn't support regex in route definitions. You would need to have a single route and match the param in your loader, then render the correct UI.

Instead, I would recommend updating your URL format to something like 
```
/:category/sub/:subcategory
/:category/:slug
```

**Reply by pankaj91as** [reply]
Thanks @kiliman for you reply.

I was migrating that was bit older & reputed news sites which having huge old data with these type of URL structure.
We are checking the camptablity if we can use & migrate using remix.

There is an anothery way I found. Basically, in loader we can verify the params & based upon that we can load the layout.
But in this process it took some extra Memory & CPU consumption that I was trying to overcome.

I was really impressed after tryed remix in few of my sites where I can handle routing by myself completely.
Really feel that if we have regex based something (either in config or server.js level) to identify URL path that would be chery on top.

Thanks again.

**Reply by kiliman** [reply]
React Router handles the routing in Remix. I know that's something that others have requested there as well especially since previous versions of RR had regex-based routing.

https://github.com/remix-run/react-router/discussions/9844
